File: projects/m5/002-polygon-area-calculator/solution/valentinaBazzacchi/shape-calculator.js
class Rectangle {
    constructor(width, height) {
        this.width = width;
        this.height = height;
    }
    // Getters and Setters for width and height parameters
    get width() {
        return this._width;
    }

    set width(newWidth) {
        this._width = newWidth;
    }

    get height() {
        return this._height;
    }

    set height(newHeight) {
        this._height = newHeight;
    }
    
    // Class Methods
    getArea() {
        return this.width * this.height;
    }

    getPerimeter() {
        return 2 * this.width + 2 * this.height;
    }

    getDiagonal() {
        return (this.width ** 2 + this.height ** 2) ** 0.5;
    }

    getPicture() {
        if (this.width > 50 || this.height > 50) return "Too big for picture.";

        let picture = "";

        for (let y = 0; y < this.height; y++) {
            for (let x = 0; x < this.width; x++) {
                picture += "*";
            }
            picture += "\n";
        }
        return picture;
    }

    getAmountInside(shape) {
        return (shape.width > this.width || shape.height > this.height) ? 0 : Math.floor(this.getArea() / shape.getArea());
    }

    string() {
        return `Rectangle(width=${this.width}, height=${this.height})`;
    }

} // end class Rectangle

class Square extends Rectangle {
    constructor(side) {
        super(side, side);
    }

    get side() {
        return newSide;
    }

    set side(newSide) {
        this.height = newSide;
        this.width = newSide;
    }

    string() {
        return `Square(side=${this.width})`;
    }
} // end subclass Square

module.exports = { Rectangle, Square };

File: projects/m5/002-polygon-area-calculator/solutions/ericadegiovanni/shape_calculator.py

class Rectangle:

    def __init__(self, width, height):

        self.width = width
        self.height = height
        

    def set_width(self, new_width):
        self.width = new_width

    def set_height(self, new_height):
        self.height = new_height

    def get_area(self):
        return self.width * self.height

    def get_perimeter(self):
        return 2 * self.width + 2 * self.height
    
    def get_diagonal(self):
        return (self.width ** 2 + self.height ** 2) ** .5

    def get_picture(self):
        if self.width <= 50 and self.height <= 50:
            line = '*' * self.width + '\n'
            return line * self.height
        else:
            return 'Too big for picture.'

    def get_amount_inside(self, shape):
        amount = self.get_area() // shape.get_area()
        return amount

    def __str__(self):
        return f'{self.__class__.__name__}(width={self.width}, height={self.height})'


class Square(Rectangle):

    def __init__(self, side):
        self.side = side
        super().__init__(width = side, height = side)
    
    def set_side(self, new_side):
        self.side = new_side
        self.height = new_side
        self.width = new_side

    def set_height(self, new_side):
        self.side = new_side
        self.height = new_side
        self.width = new_side
          
    def set_width(self, new_side):
        self.side = new_side
        self.height = new_side
        self.width = new_side
        
    def __str__(self):
        return f'{self.__class__.__name__}(side={self.side})'